sklearn/multioutput.py

```diff
@@ -359,3 +362,234 @@ def score(self, X, y):
+                col_idx = X.shape[1] + chain_idx
+                cv_result = cross_val_predict(
+                    self.base_estimator, X_aug[:, :col_idx],
+                    y=y, cv=self.cv)
+                if sp.issparse(X_aug):
+                    X_aug[:, col_idx] = np.expand_dims(cv_result, 1)
+                else:
+                    X_aug[:, col_idx] = cv_result
+
+            self.classes_.append(estimator.classes_)
+        return self
+
+    def predict(self, X):
+        """Predict on the data matrix X using the ClassifierChain model.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+
+        Returns
+        -------
+        Y_pred : array-like, shape (n_samples, n_classes)
+        """
+        X = check_array(X, accept_sparse=True)
+        Y_pred_chain = np.zeros((X.shape[0], len(self.estimators_)))
+        for chain_idx, estimator in enumerate(self.estimators_):
+            previous_predictions = Y_pred_chain[:, :chain_idx]
+            if sp.issparse(X):
+                if chain_idx == 0:
+                    X_aug = X
+                else:
+                    X_aug = sp.hstack((X, previous_predictions))
+            else:
+                X_aug = np.hstack((X, previous_predictions))
+            Y_pred_chain[:, chain_idx] = estimator.predict(X_aug)
+
+        inv_order = np.empty_like(self.order_)
+        inv_order[self.order_] = np.arange(len(self.order_))
+        Y_pred = Y_pred_chain[:, inv_order]
+
+        return Y_pred
+
+    @if_delegate_has_method('base_estimator')
+    def predict_proba(self, X):
+        """Predict probability estimates.
+
+        Probability estimates will be used as features by subsequent models
+        in the chain.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+
+        Returns
+        -------
+        Y_prob : array-like, shape (n_samples, n_classes)
+        """
+        X = check_array(X, accept_sparse=True)
+        Y_prob_chain = np.zeros((X.shape[0], len(self.estimators_)))
+        Y_pred_chain = np.zeros((X.shape[0], len(self.estimators_)))
+        for chain_idx, estimator in enumerate(self.estimators_):
+            previous_predictions = Y_pred_chain[:, :chain_idx]
+            if sp.issparse(X):
+                X_aug = sp.hstack((X, previous_predictions))
+            else:
+                X_aug = np.hstack((X, previous_predictions))
+            Y_prob_chain[:, chain_idx] = estimator.predict_proba(X_aug)[:, 1]
+            Y_pred_chain[:, chain_idx] = estimator.predict(X_aug)
+        inv_order = np.empty_like(self.order_)
+        inv_order[self.order_] = np.arange(len(self.order_))
+        Y_prob = Y_prob_chain[:, inv_order]
+
+        return Y_prob
+
+    @if_delegate_has_method('base_estimator')
+    def decision_function(self, X):
+        """Distance of the samples X to the separating hyperplane.
```

Comment:
I don't know that this is accurate for all decision_function implementations.

More accurate would be to define this in terms of the same method of the classifiers in the chain